SourceX/storm/storm.cpp

```diff
@@ -309,15 +309,19 @@ HWND SDrawGetFrameWindow(HWND *sdraw_framewindow)
 void *SMemAlloc(unsigned int amount, char *logfilename, int logline, int defaultValue)
 {
 	// fprintf(stderr, "%s: %d (%s:%d)\n", __FUNCTION__, amount, logfilename, logline);
-	assert(amount != -1u);
-	return malloc(amount);
+	//assert(amount != -1u);
+	if (amount)
+		return malloc(amount);
+	else
+		printf("%s: %d (%s:%d)\n", __FUNCTION__, amount, logfilename, logline);
 }
 
 BOOL SMemFree(void *location, char *logfilename, int logline, char defaultValue)
 {
 	// fprintf(stderr, "%s: (%s:%d)\n", __FUNCTION__, logfilename, logline);
 	assert(location);
-	free(location);
+	if (location)

```

Comment:
As this comes right after an assert the application should have failed at this point, so I don't think there is any point for this change.

Reply:
it was probably only for testing